MATLAB/DFA/src/EDA/exper_data.m

```diff
@@ -0,0 +1,360 @@
+xtickangle(45);
+title('Coefficient of Variation Values - TU9802 Amino Acids')
+legend('location', 'bestoutside');
+
+subplot(2,1,2);
+parallelcoords(CV_HCT116(aminoAcidIndex,:), 'group', metabolomics(aminoAcidIndex,1), 'Labels', all_conditions);
+xtickangle(45);
+title('Coefficient of Variation Values - HCT116 Amino Acids')
+legend('location', 'bestoutside');
+hold off
+%Now we will look at nucleosides
+nucleosideIndex = findClusterType(metabolomics, 'nucleoside');
+
+figure (13);
+hold on
+subplot(2,1,1);
+parallelcoords(CV_TU8902(nucleosideIndex,:), 'group', metabolomics(nucleosideIndex,1), 'Labels', all_conditions);
+xtickangle(45);
+title('Coefficient of Variation Values - TU9802 nucleoside')
+legend('location', 'bestoutside');
+
+subplot(2,1,2);
+parallelcoords(CV_HCT116(nucleosideIndex,:), 'group', metabolomics(nucleosideIndex,1), 'Labels', all_conditions);
+xtickangle(45);
+title('Coefficient of Variation Values - HCT116 nucleoside')
+legend('location', 'bestoutside');
+hold off
+%Now we will look at nucleotides
+nucleotideIndex = findClusterType(metabolomics, 'nucleotide');
+
+figure (14);
+hold on
+subplot(2,1,1);
+parallelcoords(CV_TU8902(nucleotideIndex,:), 'group', metabolomics(nucleotideIndex,1), 'Labels', all_conditions);
+xtickangle(45);
+title('Coefficient of Variation Values - TU9802 nucleotide')
+legend('location', 'bestoutside');
+
+subplot(2,1,2);
+parallelcoords(CV_HCT116(nucleotideIndex,:), 'group', metabolomics(nucleotideIndex,1), 'Labels', all_conditions);
+xtickangle(45);
+title('Coefficient of Variation Values - HCT116 nucleotide')
+legend('location', 'bestoutside');
+hold off
+
+%% DFA Filter 1: CV thresholds
+[metabolite_keep_TU8902, metabolite_delete_TU8902] = all_CV_Filter(CV_TU8902,metabolomics,Tabl_average_TU8902,0.9);
+[metabolite_keep_HCT116, metabolite_delete_HCT116] = all_CV_Filter(CV_HCT116,metabolomics,Tabl_average_HCT116,0.9); 
+
+%% DFA Filter 2: NaN metabolomics
+%This filter will find all of the metabolites that have no corresponding averages: no data for any of the 12 conditions. This filter will parse through the averages array and if every single column for that specific row equals 0, then the metabolite name will be added to the a new array. 
+% array for TU8902 and HCT116
+no_data_NAN_TU9802 = [];
+no_data_NAN_HCT116 = [];
+for i = 1:226
+    if mean_metabolomic_array_TU8902(i,:) == 0
+        no_data_NAN_TU9802 = [no_data_NAN_TU9802; metabolomics(i)];
+    end
+    
+    if mean_metabolomic_array_HCT116(i,:) == 0
+        no_data_NAN_HCT116 = [no_data_NAN_HCT116; metabolomics(i)];
+    end
+end
+
+%% Use Filter 2 to create new data sheet
+%Now, we will use the Filter 2 to create an excel sheet for DFA. 
+%We will first locate all the locations that have only NaN's in tables and then delete the NaN rows. The HCT116 cell line should delete 7 rows while the TU8902 cell line should delete 6 rows. 
+for i = 1:219

```

Comment:
So this doesn't work because across your 12 conditions, there are none that are all 0s. So the way I would implement this (for HCT116) is the following code:

```
% Construct an N x M boolean matrix that finds NaNs for each element
bool = isnan(table2array(Tabl_average_HCT116(:, 2:end)));

% If there are more than 6 empty values row-wise, then we'll remove the metabolite. Of course, if you find that there are still a lot NaNs, you can make this filter more stringent.
Tabl_average_HCT116(sum(double(bool), 2) >= 6, :) = [];
```

Reply:
This also doesn't involve for loops, so the code is way more elegant - just two lines.